tests/json_schema.go

```diff
@@ -158,6 +158,8 @@ func TestDataAgainstProcessor(t *testing.T, p processor.Processor, testData []Sc
 		data, err := loader.LoadData(d.File)
 		assert.Nil(t, err)
 		err = p.Validate(data)
-		assert.Contains(t, err.Error(), d.Error)
+		if assert.Error(t, err) {
```

Comment:
I generally agree that the tests are duplicates. However, the removed tests did check that `p.Validate(data)` returns an error. Removing those tests and changing this check to `if assert.Error(t, err).. ` we lose the check that the validation fails.

Reply:
That is confusing.  It doesn't actually lost the check that that validation fails, it just prevents the `assert.Contains` from panicking when err is actually nil.

Reply:
```An error is expected but got nil.``` is produced in that case

Reply:
ah sry didn't get that.